### ISSUE TITLE
define `warning` field as a child of `appMetadata` 

### ISSUE THREAD

**Issue by keighrim** [clamsproject/mmif#188]
While discussing https://github.com/clamsproject/clams-python/issues/101, we realized we can, along with `error` messages, encode `warning` messages from an app and propagate them to the users for, for example, informing a better app configuration. 

Few things to consider; 

1. unlike `error`, an app can produce multiple `warning`s. 
2. apps should be able to finish their processing and populate `annotations` list while producing `warning`s

The first note can be relatively easily addressed. However, for the second, we need to decide whether

1. an app can push annotations and related warnings into a single view OR
2. an app must add a new view that has all the warnings but an empty annotations list (remember that we allow apps to generate two or more views during its processing). 

Pursuing the first, we can have warnings closely placed to the source annotations they originated, however, we should also consider that not all warnings can be related to a specific annotation (or a set of annotations) (e.g. an undefined parameter is handed over to the app). Also having all warnings in a single view without other annotations gives us an easier way to access all warnings, so that we can, for example, wrap all warnings in an HTTP header and send them back in a response (not just buried in the MMIF payload) for better visibility. 

I want to hear what others think about the last issue.

**Comment by keighrim** [comment]
Another important decision to make is the data structure of warnings. In the error object, we included stack trace in the MMIF, so that if a user encounters an error, they can report the error (or a bug) back to the developer of the app (whose contact should be available in the app metadata). However, with warnings, I don't think app developers should be informed of all the warning messages/scenarios, and thus we can use a much simpler structure that should be just informative enough for the end users. The current draft suggests a single-string representation of individual warnings.

**Comment by marcverhagen** [comment]
I think a view with just the warnings and no annotations may be the simplest. As Keigh mentioned, having the warnings go with the view they belong to is not always possible anyway. So we would have three kinds of views: views with annotations, views with an error and views with one or more warnings. The second and third would not have annotations and no `contains` property, which would make for a simpler change in the schema than what we were discussing last week:

```json
"oneOf": [
    { "required": [ "app", "contains"  ] },
    { "required": [ "app", "error" ] }
    { "required": [ "app", "warnings" ] } ]
```

The value of `warnings` would be a list and I agree that for now a warning could just have `message` as a required property.

**Comment by keighrim** [comment]
Closed via #190.
